sechub-cli/src/mercedes-benz.com/sechub/cli/sechubconfig.go

```diff
@@ -162,13 +168,17 @@ func adjustSourceFilterPatterns(context *Context) {
 	for i, item := range context.sechubConfig.Data.Sources {
 
 		if slices.Contains(context.sechubConfig.SecretScan.Use, item.Name) {
-			// Clear all source code patterns for secrets scans
+			// Clear upload filter for secrets scans
 			context.sechubConfig.Data.Sources[i].SourceCodePatterns =
 				adjustSourceFilterPatternsWhitelistAll(item.SourceCodePatterns, true)
 		} else if slices.Contains(context.sechubConfig.CodeScan.Use, item.Name) {
 			// Append default source code patterns for code scans
 			context.sechubConfig.Data.Sources[i].SourceCodePatterns =
 				adjustSourceFilterPatternsWhitelistAll(item.SourceCodePatterns, context.config.whitelistAll)
+		} else if slices.Contains(context.sechubConfig.LicenseScan.Use, item.Name) {
```

Comment:
wouldn't it be better to combine this into the previous "else-if" statement
the executed code is the same

Reply:
For now we can do this. Right.
But in future there may be a `adjustLicensefileFilterPatternsWhitelistAll()` function.
Currently, SecHub's only license scan tool is PDS-Xray which consumes only binary files.
In future there may be integrated source code license scanners. Their file pattern may differ.

But anyway: Currently there is no reason for the extra "else if".
-> I'll change this into one "else if" statement.